packages/core/src/combinator/slice/bounds.js

```diff
@@ -0,0 +1,32 @@
+/** @license MIT License (c) copyright 2010 original author or authors */
+
+// A slice Bounds type that narrows min values via accumulation
+// and max values via Math.min.
+// 0 <= min <= max
+// slice(min2, max2 slice(min1, max1, s)) ~ slice(min1 + min2, Math.min(max1, max2), s)
+// type Bounds = { min: number, max: number }
+
+// smart constructor that enforces bounds constraints
```

Comment:
Using proper English in the comments is okay. 😄 Capitalize sentence and end with a period.